app/services/articles/enrich_image_attributes.rb

```diff
@@ -37,22 +34,14 @@ def self.call(article)
 
         next if image.blank?
 
-        attribute_width, attribute_height = image_width_height(img)
-        img["width"] = attribute_width
-        img["height"] = attribute_height
-        img["data-animated"] = true if FastImage.animated?(image, timeout: TIMEOUT, raise_on_failure: false)
+        property = FastImage.new(image, timeout: 10)
+        img["width"], img["height"] = property.size
+        img["data-animated"] = true if property.type == :gif
       end
 
       article.update_columns(processed_html: parsed_html.to_html)
     end
 
-    def self.image_width_height(img, timeout = TIMEOUT)
-      src = img.attr("src")
-      return unless src
-
-      FastImage.size(src, timeout: timeout, raise_on_failure: false)
-    end
```

Comment:
FastImage is only used in two locations and they are in a different manner. This method is adding unnecessary indirection.